lib/IAMClient.js

```diff
@@ -284,40 +284,90 @@ class VaultClient {
     /**
      * Update Quota of an account
      *
-     * @param {string} accountName - account name
      * @param {number} quota - maximum quota for the account
+     * @param {object} options - additional creation params
+     * @param {string} [options.accountName] - account name
+     * @param {boolean} [options.ci] - test flag
      * @param {VaultClient~requestCallback} callback - callback
      * @returns {undefined}
      */
-    updateAccountQuota(accountName, quota, callback) {
-        const conv = Number.isNaN(Number.parseInt(quota, 10));
-        assert(typeof accountName === 'string' && accountName !== '',
-            'accountName is required');
-        assert(typeof quota === 'number' && !conv
-            && quota >= 0, 'Quota must be a non-negative number');
-        this.request('POST', '/', true, callback, {
+    updateAccountQuota(quota, options, callback) {
+        let accountName;
+        let ci;
+        if (options !== undefined && Object.keys(options).length !== 0) {
+            ({ accountName, ci } = options);
+        }
+        const data = {
             Action: 'UpdateAccountQuota',
             Version: '2010-05-08',
-            AccountName: accountName,
             quotaMax: quota,
-        });
+        };
+        if (ci) {
```

Comment:
should introduce a test flag in the API of the component: esp. here what this really does is simply skip validation of quota field & type of accoujnt name....

Reply:
> This allows us to test the API-side validation, in a functional way

api-side validation can be tested anyway by calling the api directly (bypassing vaultclient), even if vaultclient is used in the test for setup/...
"Functionally", vaultclient is expected to do validation : so testing the validation of the API using the tool in a tweaked mode is not really the functional case...

> And like aws cli, this introduces an option (could be a better name than ci), that disable input validation on vaultclient's side. This is similar to the parameter_validation aws cli parameter. 

The parameter is part of the API, and introduces complexity in the code : so the real question is really "is such a flag really useful", e.g. is it worth the added maintenance effort.
→ if not or if in doubt, let's just remove it, keep things simple and code lean
→ if this is really a useful feature, it MUST be renamed and define a proper semantics for the flag (like `parameter_validation`, which has a very clear semantic)

Reply:
the flag has been renamed disableValidation see here : https://github.com/scality/vaultclient/pull/442/files#diff-845cfb35dcf0685889460bfb3868ad667256bef6c8c053f7d12a06987297e75dR20-R35

Reply:
If we really want to this way, I think we should stick to `parameter_validation` (or `parameterValidation`), as it is clearer and similar to AWS

Reply:
flag name changed here : https://github.com/scality/vaultclient/pull/442/commits/8add00de87cae735118d30f71bb56465d4aac546#diff-6ed3f39879bec9944e1a6e9d0d3c1bcbc7209a7eab826e5c3819048e4841163dR92